core/query/sql.ml

```diff
@@ -293,11 +318,13 @@ class virtual printer =
           self#pp_update ppf upd_table upd_fields upd_where
       | Insert { ins_table; ins_fields; ins_records } ->
           self#pp_insert ppf ins_table ins_fields ins_records
-      | With (z, q, q') ->
+      | With (z, q, qs) ->
           Format.fprintf ppf "with %s as (@[<v>%a@])\n%a"
             z
             pr_q q
-            pr_q q'
+            pp_qs qs
+      | Transaction qs ->
+          Format.fprintf ppf "BEGIN; %a COMMIT;" pp_qs qs
```

Comment:
Here of course is a place where in general a DB might return an exception/error because of concurrent write activity that committed first.  I guess currently if this ever happens we will just get a hard run time failure. Of course, standalone update operations can fail too for example by violating integrity constraints.  Sounds like a great use case for effect / exception handlers.

Reply:
Ah, you read my mind! Yes, indeed, currently this will net you a runtime failure (like with the lens code). 
Maybe we should think of working out a proper design for #565.  I think it's outside of the scope of this patch though.

Reply:
Agreed.